_posts/2023-17-10-remote-attestation-blog-part1.markdown

```diff
@@ -0,0 +1,139 @@
+
+Secure Boot prevents boot if the signature cannot be validated against a certificate enrolled in the bootchain. For example if someone changes the kernel to a custom one, it is not signed by the OS vendor. This causes the boot to stop because the bootloader couldn't verify the signature. The process of checking the kernel's signatures occurs before the OS ever runs.
+
+<h3 style="text-align left; color:#6B98BF;"> What is Measured Boot? </h3>
+
+Unlike Secure Boot, Measured Boot will measure each startup component, including firmware, all the way up to the boot drivers. A hash is taken at the first step in the boot process and is extended for the next object(s) in the chain. It is stored in a chosen Platform Configuration Register (PCR), which is a memory location in the TPM. The extend operation allows for data to be appended the value already stored in the PCR. A hash of the newly formed data is the result, and that is stored back into the PCR. The final hash can be seen as a checksum of log events. With this, an auditor can later come and validate the logs by computing the expected PCR values from the log and then comparing them to the PCR values of the TPM. One can verify the validity of the kernel because measured boot checks each compenent in the start-up process and the final hash value encapsulates all of them in their present (at that time) state. With Measured Boot, the boot process is never stopped, but it provides the necessary information to detect attacks. 
+
+<h3 style="text-align left; color:#6B98BF;"> Measured Boot and Secure Boot Go Hand in Hand </h3>
+
+These two processes go hand-in-hand to ensure a trusted Operating System (O/S) boots. Measured Boot guards the system from the processor powering on to the point where the operating system is ready to run. The issue is that once the O/S is booted, Measured Boot stops and its output is encapsulated into the measured boot log and the PCR values in the TPM. Now that the O/S is unguarded, it is **conceivable** that someone could manage to navigate to the boot event log, the PCRs, and the TPM and alter them retroactively. They could even go as far as to replace the TPM device with a virtual TPM. 
+
+Secure Boot depends on Measured Boot to guard the UEFI Bios. Measured Boot will check the competent and take a hash of it to be stored for later to help prove whether or not it has been tampered with. The UEFI Bios will prevent an unsigned kernel from booting properly if the hashes taken do not match at the time and will boot the kernel in “lockdown mode,” which will prevent alterations to it. This makes making retroactive alteration of the measured boot output harder. 
```

Comment:
No only secure boot does this all. It just when Secure Boot does a decision it gets measured by measured boot.

>This makes making retroactive alteration of the measured boot output harder.

Yes, but the TPM is the reason that it doesn't matter, not secure boot.

Secure Boot -> Local integrity checking at the point of execution
Measured Boot -> Allows checking the integrity after the system was bootet.

Reply:
Should I take the last sentence out since it isn't accurate or rephrase it to include the reason being the TPM?

Reply:
I can also add a sentence after the ***which will prevent alterations to it.*** to say that measured boot will allow for integrity checking after the system is booted